pkg/devicetwin/dtcontext/dtcontext_test.go

```diff
@@ -0,0 +1,493 @@
+package dtcontext
+
+import (
+	"encoding/json"
+	"errors"
+	"reflect"
+	"sync"
+	"testing"
+
+	"github.com/kubeedge/kubeedge/beehive/pkg/common/config"
+	"github.com/kubeedge/kubeedge/beehive/pkg/core"
+	"github.com/kubeedge/kubeedge/beehive/pkg/core/context"
+	"github.com/kubeedge/kubeedge/beehive/pkg/core/model"
+	"github.com/kubeedge/kubeedge/pkg/common/util"
+	"github.com/kubeedge/kubeedge/pkg/devicetwin/dtcommon"
+	"github.com/kubeedge/kubeedge/pkg/devicetwin/dttype"
+)
+
+// TestInitDTContext is function to test InitDTContext().
+func TestInitDTContext(t *testing.T) {
+	util.LoadConfig()
+	nodeID, err := config.CONFIG.GetValue("edgehub.controller.node-id").ToString()
+	if err != nil {
+		t.Errorf("Error in getting node id %v", err)
+		return
+	}
+	tests := []struct {
+		name    string
+		context *context.Context
+		want    *DTContext
+		wantErr error
+	}{
+		{
+			name:    "ActualContextArgumentTest",
+			context: context.GetContext(context.MsgCtxTypeChannel),
+			want: &DTContext{
+				NodeID:         nodeID,
+				CommChan:       make(map[string]chan interface{}),
+				ConfirmChan:    make(chan interface{}, 1000),
+				ModulesContext: context.GetContext(context.MsgCtxTypeChannel),
+				State:          dtcommon.Disconnected,
+			},
+			wantErr: nil,
+		},
+		{
+			name:    "EmptyContextArgumentTest",
+			context: context.GetContext("test"),
+			want: &DTContext{
+				NodeID:         nodeID,
+				CommChan:       make(map[string]chan interface{}),
+				ConfirmChan:    make(chan interface{}, 1000),
+				ModulesContext: context.GetContext("test"),
+				State:          dtcommon.Disconnected,
+			},
+			wantErr: nil,
+		},
+	}
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			got, err := InitDTContext(test.context)
+			if err != test.wantErr {
+				t.Errorf("InitDTContext() error = %v, wantErr %v", err, test.wantErr)
+				return
+			}
+			if !reflect.DeepEqual(got.NodeID, test.want.NodeID) {
+				t.Errorf("NodeID = %v, want = %v", got.NodeID, test.want.NodeID)
```

Comment:
Can we reflect.DeepEqual of want and got directly instead of all fields individually ?